.github/workflows/root-ci-config/build_root.py

```diff
@@ -222,23 +250,23 @@ def cleanup_previous_build():
 
 
 @github_log_group("Pull/clone branch")
-def git_pull(repository: str, branch: str):
+def git_pull(directory: str, repository: str, branch: str):
     returncode = 1
 
     for _ in range(5):
         if returncode == 0:
             break
 
-        if os.path.exists(f"{WORKDIR}/src/.git"):
+        if os.path.exists(f"{WORKDIR}/{directory}/.git"):
             returncode = subprocess_with_log(f"""
-                cd '{WORKDIR}/src'
+                cd '{WORKDIR}/{directory}'
```

Comment:
another candidate for `os.path.join`

Reply:
Do you have a 'good' way to use it inside this long string?

Reply:
I believe something like this should work
```python
cd '{os.path.join(f"{WORKDIR}", f"{directory}")}'
```